maven-plugin-tools-java/src/main/java/org/apache/maven/tools/plugin/extractor/javadoc/JavaJavadocMojoDescriptorExtractor.java

```diff
@@ -510,7 +510,7 @@ private Map<String, JavaField> extractFieldParameterTags(JavaClass javaClass) {
         if (superClass != null) {
             rawParams = extractFieldParameterTags(superClass);
         } else {
-            rawParams = new TreeMap<String, JavaField>();
+            rawParams = new TreeMap<>();
```

Comment:
Looking at this one, I realize the diamond operator with no params is more useful for the case where it's declared on the same line. 

TreeMap<String, JavaField> rawParams = new TreeMap<String, JavaField>();

does seem redundant

rawParams = new TreeMap<String, JavaField>();

is not. Haven't noticed that before.

Reply:
Doesn't matter that much. The variable declaration wasn't in view by default in Github code review. I do think there's a real tendency for devs to argue that, "In my tool it's obvious" when advocating for more compact, less redundant syntax without really considering how it looks in other environments. I often hear folks claim that IDEs make certain constructs unnecessary. We can be really myopic about that, even to the point of saving a few milliseconds of typing right now at the cost of hours of debugging to our future selves. The real cost here is in lambdas and method chaining, which are extremely opaque and hard to read, all to save the trivial effort of typing a few more local variables.

Reply:
I'll keep an eye out of future PRs for this pattern, although I think we've gone through the vast majority by now. There's a  couple projects left with one or two cases, which I hope to pick up along with other more meaningful changes.